frontend/cypress/integration/tests/404.spec.js

```diff
@@ -0,0 +1,26 @@
+describe('Visits homepage', () => {
+  it('Visits the CloudHaven Homepage', () => {
+    cy.visit('http://localhost:3000');
+    cy.contains('CloudHaven');
+    cy.contains('Login');
+  });
+});
+
+describe('Visits invalid page, checks for 404', () => {
```

Comment:
You should also include a test that makes sure it redirects to the homepage when you click "back to home" on the 404 page.